api/src/main/java/uk/gov/hmcts/payment/api/controllers/ReplayCreditAccountPaymentController.java

```diff
@@ -0,0 +1,191 @@
+import io.swagger.annotations.ApiResponses;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.payment.api.configuration.LaunchDarklyFeatureToggler;
+import uk.gov.hmcts.payment.api.contract.*;
+import uk.gov.hmcts.payment.api.contract.util.CurrencyCode;
+import uk.gov.hmcts.payment.api.contract.util.Service;
+import uk.gov.hmcts.payment.api.model.PaymentStatus;
+import uk.gov.hmcts.payment.api.service.ReplayCreditAccountPaymentService;
+import uk.gov.hmcts.payment.api.v1.model.exceptions.PaymentException;
+import uk.gov.hmcts.payment.api.v1.model.exceptions.PaymentNotFoundException;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@Api(tags = {"Replay Credit Account Payment"})
+public class ReplayCreditAccountPaymentController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ReplayCreditAccountPaymentController.class);
+
+    private static final String FAILED = "failed";
+
+
+    private final LaunchDarklyFeatureToggler featureToggler;
+
+    private final ReplayCreditAccountPaymentService replayCreditAccountPaymentService;
+    private final CreditAccountPaymentController creditAccountPaymentController;
+
+    private final static String PAYMENT_STATUS_HISTORY_MESSAGE = "System Failure. Not charged";
+
+
+    @Autowired
+    public ReplayCreditAccountPaymentController(LaunchDarklyFeatureToggler featureToggler,
+                                                ReplayCreditAccountPaymentService replayCreditAccountPaymentService,
+                                                CreditAccountPaymentController creditAccountPaymentController) {
+        this.featureToggler = featureToggler;
+        this.replayCreditAccountPaymentService = replayCreditAccountPaymentService;
+        this.creditAccountPaymentController = creditAccountPaymentController;
+    }
+
+    @ApiOperation(value = "Replay credit account payment", notes = "Replay credit account payment")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Replay Payment Completed Successfully"),
+        @ApiResponse(code = 400, message = "BAD Request"),
+        @ApiResponse(code = 500, message = "Replay Payment failed")
+    })
+    @PostMapping(value = "/replay-credit-account-payments", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    @ResponseBody
+    @Transactional
+    public ResponseEntity<String> replayCreditAccountPayment(@RequestParam("csvFile") MultipartFile replayPBAPaymentsFile,
```

Comment:
Most of the logic needs to be moved to service layer. it is ok to go ahead for now because of limited time we have, but this needs to refactored at later time

Reply:
Yes we will move logic to service layer later